ext/standard/string.c

```diff
@@ -1033,6 +1033,15 @@ PHP_FUNCTION(implode)
 		Z_PARAM_ARRAY_HT_OR_NULL(pieces)
 	ZEND_PARSE_PARAMETERS_END();
 
+	if (arg1_str != NULL && pieces == NULL) {
+		zend_type_error(
```

Comment:
Can't you just adjust the "Argument #1 ($array) must be of type array, string given" message below? It throws under the exact same conditions (`pieces == NULL` and `arg1_str != NULL` i.e. `arg1_array == NULL`).

Reply:
So that second message is now unreachable now and should be removed, correct?

I changed it. Let me know if that's what you expected.

Reply:
You should keep the original if/else structure and just change the error message. The original code eliminates some redundant runtime checks. We generally discourage refactoring that doesn't have a clear benefit.

Reply:
Which redundant runtime checks are done now that were eliminated in the original structure? I wasn't able to identify it. 😳

Reply:
The original code checks for `pieces == NULL` once, the new code does so 3 times.